ext/bcmath/libbcmath/src/init.c

```diff
@@ -82,6 +82,20 @@ void bc_init_numbers(void)
 	BCG(_two_)->n_value[0] = 2;
 }
 
+static void _bc_force_free_number(bc_num *num)
+{
+	pefree((*num)->n_ptr, 1);
+	pefree(*num, 1);
+	*num = NULL;
+}
+
+void bc_force_free_numbers(void)
```

Comment:
During GSHUTDOWN you should use the passed `bcmath_globals` instead of BCG to avoid freeing the wrong data in ZTS. In fact, If I were you I'd call `_bc_force_free_number` directly in GSHUTDOWN.